sensor/common/deduper/key.go

```diff
@@ -0,0 +1,84 @@
+package deduper
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	eventPkg "github.com/stackrox/rox/pkg/sensor/event"
+)
+
+var (
+	deduperTypes = []any{
+		&central.SensorEvent_NetworkPolicy{},
+		&central.SensorEvent_Deployment{},
+		&central.SensorEvent_Pod{},
+		&central.SensorEvent_Namespace{},
+		&central.SensorEvent_Secret{},
+		&central.SensorEvent_Node{},
+		&central.SensorEvent_ServiceAccount{},
+		&central.SensorEvent_Role{},
+		&central.SensorEvent_Binding{},
+		&central.SensorEvent_NodeInventory{},
+		&central.SensorEvent_ProcessIndicator{},
+		&central.SensorEvent_ProviderMetadata{},
+		&central.SensorEvent_OrchestratorMetadata{},
+		&central.SensorEvent_ImageIntegration{},
+		&central.SensorEvent_ComplianceOperatorResult{},
+		&central.SensorEvent_ComplianceOperatorProfile{},
+		&central.SensorEvent_ComplianceOperatorRule{},
+		&central.SensorEvent_ComplianceOperatorScanSettingBinding{},
+		&central.SensorEvent_ComplianceOperatorScan{},
+		&central.SensorEvent_AlertResults{},
+	}
+)
+
+// Key is the Key by which messages are deduped.
+type Key struct {
+	ID           string
+	ResourceType reflect.Type
+}
+
+// CopyDeduperState makes a copy of the deduper state.
+func CopyDeduperState(state map[string]uint64) map[Key]uint64 {
+	if state == nil {
+		return make(map[Key]uint64)
+	}
+
+	result := make(map[Key]uint64, len(state))
+	for k, v := range state {
+		parsedKey, err := keyFrom(k)
+		if err != nil {
+			log.Warnf("Deduper state received from central has malformed entry: %s->%d: %s", k, v, err)
```

Comment:
nit: if we think of this code as a library, maybe remove the assumption that the state came from central? 
```suggestion
			log.Warnf("Deduper state has malformed entry: %s->%d: %s", k, v, err)
```

Reply:
Done